luaparse.js

```diff
@@ -1846,7 +1848,10 @@
         if (Identifier === token.type) {
           var parameter = parseIdentifier();
           // Function parameters are local.
-          if (options.scope) scopeIdentifier(parameter);
+          if (options.scope) scopeIdentifier(parameter, {
+            parameterOf: name,
```

Comment:
`name` contains either a node with the function name or `null` (for function expressions); function names, even if present, are themselves subject to scoping and therefore potentially ambiguous. For the originally given use case of code completion, it would be useful to distinguish a `local function foo()` defined on line 15 from a `function foo()` defined on line 75.